kernel: bump 6.6 to 6.6.59

namiltd opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.59

Removed upstreamed:
 - generic/backport-6.6/777-netfilter-xtables-fix-typo-causing-some-targets-to-not-load-on-IPv6.patch[1]
 - generic/backport-6.6/780-24-v6.12-r8169-avoid-unsolicited-interrupts.patch[2]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.59&id=433742ba96baf30c21e654ce3e698ad87100593b
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.59&id=7d6d46b429804b1a182106e27e2f8c0e84689e1a

Additionally, the mac80211 package has been patched for compatibility with kernel 6.6.59.

graysky2 commented:
Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

namiltd committed:
mac80211: compatibility with kernel 6.6.59

Fix genlmsg_multicast_allns() build error on kernel 6.6.59.
Based on kernel patch https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.6.59&id=e0f83d268974dab0361d11904dfc9acec53f96a6

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/16835
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented:
@graysky2 Please check the build now.

namiltd commented:
Build system: x86/64
Build-tested: ramips/tplink_archer-c5-v4
Run-tested: ramips/tplink_archer-c5-v4

graysky2 commented:
@namiltd - builds successfully

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

namiltd commented:
Ping @hauke

DragonBluep commented on package/kernel/mac80211/patches/build/236-fix-genlmsg_multicast_allns-build-error-on-kernel-6.6.59.patch:
```diff
@@ -0,0 +1,100 @@
+--- a/backport-include/net/genetlink.h
++++ b/backport-include/net/genetlink.h
+@@ -150,7 +150,7 @@ int genlmsg_multicast(const struct genl_
```
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v6.11.6&id=e0f83d268974dab0361d11904dfc9acec53f96a6

Please backport the full patch and keep the header.

namiltd replied:
The patch cannot be fully backported because it affects more files. Only changes related to files in mac80211 are imported.

namiltd committed:
kernel: bump 6.6 to 6.6.59

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.59

Removed upstreamed:
 - generic/backport-6.6/777-netfilter-xtables-fix-typo-causing-some-targets-to-not-load-on-IPv6.patch[1]
 - generic/backport-6.6/780-24-v6.12-r8169-avoid-unsolicited-interrupts.patch[2]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.59&id=433742ba96baf30c21e654ce3e698ad87100593b
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.59&id=7d6d46b429804b1a182106e27e2f8c0e84689e1a

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/16835
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

zxlhhyccc commented:
@namiltd Can it be merged? It looks messy.

namiltd commented:
@zxlhhyccc Yes. PR is complete

zxlhhyccc commented:
> @zxlhhyccc Yes, PR is complete.

@namiltd I mean can multiple submissions be combined into one?

namiltd commented:
@zxlhhyccc There are only two commits, one is about changing the kernel and the other is about making the mac80211 package work with this kernel.

danpawlik reviewed (approved):
Works well on BPi-R4 and AX3200

namiltd commented:
@hauke Please merge this into openwrt-24.10 branch as well.
